.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - deploy
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: satackey/action-docker-layer-caching@v0.0.8
+      - run: docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app
+
+  deploy:
+    needs: test
```

Comment:
you could deploy on tags by adding this line:

`
    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
`

Reply:
got it, but there's not a mapping from tags to deployment groups